### PR TITLE
chore(localdebug): support multiple ports in dev tunnel task

### PR DESCRIPTION
**Pull request by XiaofuHuang**
Support multiple ports in dev tunnel task.
Schema: 
```
{
    "label": "Start local tunnel",
    "type": "teamsfx",
    "command": "debug-start-local-tunnel",
    "args": {
        "type": "dev-tunnel",
        "ports": [
            {   
                "portNumber": 53000,
                "protocol": "https",
                "access": "private",
                "writeToEnvironmentFile": {
                    "endpoint": "TAB_ENDPOINT", 
                    "domain": "TAB_DOMAIN"
                }
            }
        ],
        "env": "local"
    },
    "isBackground": true,
    "problemMatcher": "$teamsfx-local-tunnel-watch"
}

```

TODO: 
1. change template and samples https://github.com/OfficeDev/TeamsFx/pull/8190
2. change task definitation
3. rename `output` to be `writeToEnvironmentFile` in ngrok task

### CODE REVIEW

File: packages/vscode-extension/src/debug/taskTerminal/ngrokTunnelTaskTerminal.ts [hunk 1]
```diff
@@ -57,6 +57,10 @@ export interface INgrokTunnelArgs extends IBaseTunnelArgs {
   ngrokArgs: string[];
   ngrokPath?: string;
   tunnelInspection?: string;
+  output?: {
```
Comment (kuojianlu) [inline packages/vscode-extension/src/debug/taskTerminal/ngrokTunnelTaskTerminal.ts:60]: `writeToEnvironmentFile`?

File: packages/vscode-extension/src/debug/taskTerminal/devTunnelTaskTerminal.ts [hunk 2]
```diff
@@ -133,51 +142,48 @@ export class DevTunnelTaskTerminal extends BaseTunnelTaskTerminal {
...
+          continue;
+        }
 
-      if (tunnelInstance?.tags?.includes(DevTunnelTag)) {
-        await this.tunnelManagementClientImpl.deleteTunnel(tunnelInstance);
+        const tunnelInstance = await this.tunnelManagementClientImpl.getTunnel({
+          tunnelId: devTunnelState.tunnelId,
+          clusterId: devTunnelState.clusterId,
+        });
+        if (tunnelInstance?.tags?.includes(DevTunnelTag)) {
```
Comment (swatDong) [inline packages/vscode-extension/src/debug/taskTerminal/devTunnelTaskTerminal.ts:156]: Delete tunnel state when `tunnelInstance` is undefined?

File: packages/vscode-extension/src/debug/taskTerminal/utils/devTunnelStateManager.ts [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+  private static mutex = withTimeout(new Mutex(), 1000);
+  private readonly devTunnelStateKey = "teamsToolkit:devtunnel";
+  private stateService: IStateService;
+
+  constructor(stateService: IStateService) {
+    this.stateService = stateService;
+  }
+
+  public static create(): DevTunnelStateManager {
+    const stateService = isFeatureFlagEnabled(FeatureFlags.DevTunnelTest)
```
Comment (swatDong) [inline packages/vscode-extension/src/debug/taskTerminal/utils/devTunnelStateManager.ts:33]: Why adding this flag?

File: packages/vscode-extension/src/debug/taskTerminal/ngrokTunnelTaskTerminal.ts (hunk 1 above)
Comment (XiaofuHuang) [reply packages/vscode-extension/src/debug/taskTerminal/ngrokTunnelTaskTerminal.ts:60]: The ngrok output change will be in another PR.

File: packages/vscode-extension/src/debug/taskTerminal/utils/devTunnelStateManager.ts (hunk 3 above)
Comment (XiaofuHuang) [reply packages/vscode-extension/src/debug/taskTerminal/utils/devTunnelStateManager.ts:33]: It is used for UI test to clean the basis tunnel. The `vscode.context.workspaceState` is saved in sqllite db and hard to get in the UI test.

Comment (swatDong) [reply packages/vscode-extension/src/debug/taskTerminal/utils/devTunnelStateManager.ts:33]: Ok, for test only. Make sense.

**Review by swatDong — APPROVED** [review]

File: packages/vscode-extension/src/debug/taskTerminal/devTunnelTaskTerminal.ts (hunk 2 above)
Comment (XiaofuHuang) [reply packages/vscode-extension/src/debug/taskTerminal/devTunnelTaskTerminal.ts:162]: Updated. 
Note: if the user switch the m365 account, the `tunnelInstance` will be `undefined` and the tunnel will be delete by dev tunnel service after 30 days.

**Review by kuojianlu — APPROVED** [review]

**Review by HuihuiWu-Microsoft — APPROVED** [review]

**Pull request merged**